### PR TITLE
Add none body and header support

### PR DESCRIPTION
**Pull request by bbangert**
Two missing features I noticed when using with a yaml file that did header verification. I was trying to return a Response of 201 with a Location header set, no body, and these changes were required before things worked right.

### CODE REVIEW

**Comment by djeebus** [comment]
Nice, thanks! Can we get some tests proving that these things were broken before, but fixed now?

**Comment by bbangert** [comment]
Well, the headers one is pretty obvious if you have a schema requiring it, since bravado-core looks for the headers attribute. I have an app which needed these changes here: https://github.com/mozilla-services/push-messages/ though that obviously isn't a test-case. :)

I'll try and get some tests whipped up shortly.

**Comment by djeebus** [comment]
Thanks!

**Comment by bbangert** [comment]
Ok, maybe not, I don't know how to make the existing tests pass, nor is there any instructions about how to run them. I saw a tox.ini, so I tried tox and got a ton of fail. Suggestions?

**Comment by djeebus** [comment]
`tox` is definitely how you should be running the tests. They all pass on my ubuntu 15.10 machine w/ tox 2.3.1. What error message are you getting?

**Comment by bbangert** [comment]
I get this: 

```
ERROR:   py26: InterpreterNotFound: python2.6
ERROR:   py27: commands failed
ERROR:   py33: InterpreterNotFound: python3.3
ERROR:   py34: commands failed
ERROR:   py35: commands failed
ERROR:   py27-pyramid15: commands failed
ERROR:   py35-pyramid15: commands failed
ERROR:   pep8: commands failed
```

**Comment by bbangert** [comment]
There's a ton more errors over a gazillion pages worth of scrollback.

Most of them are about:

```
E           FileNotFoundError: [Errno 2] No such file or directory: 'setuptools/tests/svn_data/svn17_ext_list.xml'
```

or related errors. I have subversion installed, not sure what is required to make this dir show up.

**Comment by djeebus** [comment]
`ERROR:   py26: InterpreterNotFound: python2.6` is pretty self explanatory; python2.6 isn't installed! we'll skip that :)

TL;DR: move your virtualenv out of your src folder, or edit `tox.ini` as below.

The rest of the errors look like they're caused by the fact that `pyramid_swagger`'s `tox.ini` doesn't point to a single folder full of tests; it relies on `py.test` to figure out where the tests are on it's own. It does that by spidering the current folder _and all subfolders_ to find anything that starts with the word `test`. I'm assuming that you created a virtual environment in the same folder you checked out `pyramid_swagger`, and it's finding all the tests that every module installed with their project (including `setuptools`). 

If you edit your `tox.ini` file so that the `commands` key under `[testenv]` looks like the following, it should work too (the key is to add `tests/` after `--strict`):

```
[testenv]
# ... snipped for brevity ... 
commands =
    coverage run --source=pyramid_swagger/ --omit=pyramid_swagger/__about__.py -m pytest --capture=no --strict tests/ {posargs}
    coverage report -m
```

**Comment by bbangert** [comment]
Ah, its so common to install the virtualenv in the project dir, it hadn't occurred to me that it shouldn't be done. Is this noted in docs somewhere?

**Comment by bbangert** [comment]
Moving the virtualenv out did fix it, I'll get a few tests in for this now. Thanks.

**Comment by djeebus** [comment]
@bbangert no worries :) I've ran into this a few times at work. Most open source projects I've ran across pass a tests folder to their runner explicitly, but the ones at work are hit or miss; I recognized the symptoms.

I've created a PR to fix this on `pyramid_swagger`: striglia/pyramid_swagger#157
